src/js/forms/sagas.js

```diff
@@ -0,0 +1,22 @@
+import { put, select, takeLatest, throttle } from "redux-saga/effects";
+import { DELETE_PERSISTENT_FORM_STATE, SET_PERSISTENT_FORM_STATE } from "../app/actionTypes";
+import { setSessionStorage } from "../utils/utils";
+import { updateReduxFormState } from "./actions";
+import { getAllFormState } from "./selectors";
+
+export function* setPersistentFormState({ payload: { formName, formValues } }) {
+    yield put(updateReduxFormState(formName, formValues));
+    const formState = yield select(getAllFormState);
+    setSessionStorage("formState", formState);
+}
+
+export function* deletePersistentFormState({ payload: { formName } }) {
+    yield put(updateReduxFormState(formName, null));
+    const formState = yield select(getAllFormState);
+    setSessionStorage("formState", formState);
+}
```

Comment:
How about:
1. Pull current state from Redux.
2. Merge update into current state in saga.
3. Update localStorage.
4. Update Redux.